pkg/gardenlet/controller/seed/seed/components.go

```diff
@@ -97,14 +97,26 @@ func defaultIstio(
 		return nil, err
 	}
 
+	gardenSeed := seed.GetInfo()
+
+	var minReplicas *int
+	var maxReplicas *int
+	if len(gardenSeed.Spec.Provider.Zones) > 1 {
+		minReplicas = pointer.Int(len(gardenSeed.Spec.Provider.Zones) * 2)
+		maxReplicas = pointer.Int(len(gardenSeed.Spec.Provider.Zones) * 4)
```

Comment:
With 3 zones, we have at least 6 replicas and at most 12? Is this expected? Can you explain/add a comment here why this is done this way? It's not quite obvious when reading this code.

Reply:
In the past, istio ingress gateway was autoscaled with 2-5 replicas, with minimum/maximum being statically defined in the manifests.
On some infrastructures, e.g. AWS, a loadbalancer spanning multiple availability zones may get an individual IP address per availability zone. I intended to have at least two replicas per availability zone for availability reasons. Otherwise, one (unlucky) node error may become directly visible to external consumers with a probability of 1/#availability zones. This is mitigated by having a minimum of two replicas per zone.
As we had no problems with a maximum of 5 replicas in the past, I did not want to scale up to 5 replicas per zone and "just" allowed a maximum of 4 replicase per zone. In effect this can theoretically lead to a much higher amount of istio replicas compared to the past (12 vs. 5). However, as we have not seen problems with at most 5 replicas I am confident that we should rarely reach 12 replicas at all.

Is this explanation sufficient or do you want me to add it into the code as well?

Reply:
Interesting 😄 Yeah, well, I think a comment here in the code is reasonable since it's not obvious where these numbers are coming from :)

Reply:
Added corresponding comment with latest commit.